### ISSUE TITLE
autokey regularly blocks keyboard 

### ISSUE THREAD

**Issue by herrdeh** [autokey/autokey#225]
## Classification:

crash
## Reproducibility:

often

## Summary

after calling several phrases, AK blocks the keyboard

## Steps to Reproduce (if applicable)

- open a text window, eg in evolution or gedit
- type several abreviations

## Expected Results

- phrases should expand

## Actual Results

- 1-3 phrases do expand, then no longer
- all keyboard inputs are blocked
- I can get out by closing the text window with mouse click, killall -9 autokey-gtk, relaunch autokey by using the gnome launcher
- if I switch to another app with the mouse, the whole computer gets blocked and needs a cold reboot, no way to get to a terminal

This comes from the terminal with AK launched by "autokey-gtk --verbose"
```
2018-12-04 07:20:22,675 DEBUG - service - Raw key: 's', modifiers: [], Key: s
2018-12-04 07:20:22,675 DEBUG - service - Window visible title: '*Ungespeichertes Dokument 1 - gedit', Window class: 'gedit.Gedit'
2018-12-04 07:20:22,675 DEBUG - service - No phrase/script matched hotkey
2018-12-04 07:20:22,677 DEBUG - service - Input stack at end of handle_keypress: deque(['\n', 's'], maxlen=150)
2018-12-04 07:20:22,843 DEBUG - service - Raw key: 'g', modifiers: [], Key: g
2018-12-04 07:20:22,843 DEBUG - service - Window visible title: '*Ungespeichertes Dokument 1 - gedit', Window class: 'gedit.Gedit'
2018-12-04 07:20:22,843 DEBUG - service - No phrase/script matched hotkey
2018-12-04 07:20:22,846 DEBUG - service - Input stack at end of handle_keypress: deque(['\n', 's', 'g'], maxlen=150)
2018-12-04 07:20:22,923 DEBUG - service - Raw key: 'e', modifiers: [], Key: e
2018-12-04 07:20:22,923 DEBUG - service - Window visible title: '*Ungespeichertes Dokument 1 - gedit', Window class: 'gedit.Gedit'
2018-12-04 07:20:22,923 DEBUG - service - No phrase/script matched hotkey
2018-12-04 07:20:22,926 DEBUG - service - Input stack at end of handle_keypress: deque(['\n', 's', 'g', 'e'], maxlen=150)
2018-12-04 07:20:23,017 DEBUG - service - Raw key: <Key.ENTER: '<enter>'>, modifiers: [], Key: Key.ENTER
2018-12-04 07:20:23,018 DEBUG - service - Window visible title: '*Ungespeichertes Dokument 1 - gedit', Window class: 'gedit.Gedit'
2018-12-04 07:20:23,018 DEBUG - service - No phrase/script matched hotkey
2018-12-04 07:20:23,019 DEBUG - iomediator - Send via clipboard
2018-12-04 07:20:23,019 DEBUG - interface - Sending string via clipboard: Sehr geehrte Frau

2018-12-04 07:20:23,019 DEBUG - service - Input stack at end of handle_keypress: deque([], maxlen=150)
2018-12-04 07:20:23,019 DEBUG - interface - Sending via clipboard enqueued.
2018-12-04 07:20:23,020 ERROR - service - Ignored locking error in handle_keypress
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/autokey/service.py", line 206, in __tryReleaseLock
    self.configManager.lock.release()
RuntimeError: release unlocked lock
2018-12-04 07:20:23,020 DEBUG - interface - Send special key: [<Key.BACKSPACE: '<backspace>'>]
2018-12-04 07:20:23,022 DEBUG - interface - Send special key: [<Key.BACKSPACE: '<backspace>'>]
2018-12-04 07:20:23,023 DEBUG - interface - Send special key: [<Key.BACKSPACE: '<backspace>'>]
2018-12-04 07:20:23,024 DEBUG - interface - Send special key: [<Key.BACKSPACE: '<backspace>'>]

```

## Version

AutoKey version: 
0.95.1
Used GUI (Gtk, Qt, or both): 
Gtk
Installed via: (PPA, pip3, …).
system default

Distro:
ubuntu 18.04
## Notes

Describe any debugging steps you've taken yourself.

If you've found a workaround, please provide it here.
See above - close accident window by mouse, killall -9 autokey-gtk, relaunch AK

**Comment by josephj11** [comment]
Just to clarify I assume the above log is snipped in several places. Otherwise, it would be a bit odd that  AutoKey is sending anything without also reporting that a trigger phrase or hotkey was detected.
I briefly reviewed the change logs of newer versions (not yet configured for general release) and don't see fixes that address this.

**Comment by herrdeh** [comment]
I cut out just the crash incident. Meaning: Just the trigger phrase "sge" - and what happens then. This is one sequence of output, whithout any gaps or leftouts or whatever.
Am I clear enough?

**Comment by josephj11** [comment]
I just reviewed a couple of other traces in recent updates to issues and they don't seem to have the messages I was looking for in them either. I'll ask on Gitter.

**Comment by herrdeh** [comment]
I'll be happy to provide other output from my crashes, just let me know what exactly you need.

**Comment by josephj11** [comment]
That's just a separate bug in the logging code. I mentioned it to Thomas.

**Comment by luziferius** [comment]
Hmmm. The new description reads exactly like #226. You both use the clipboard pasting method and get keyboard lock-ups.
@herrdeh I tried a bugfix that hopefully causes AutoKey to always release the keyboard after it pasted text. You may want to try it and see if this helps: https://github.com/autokey/autokey/tree/bugfix/issue_226
Either do a git clone and checkout the `bugfix/issue_226` branch or go to the location in the link above and download a ZIP archive snapshot.
Then 
1. (Unzip the archive)
2. open a terminal inside the `lib` directory in the downloaded autokey directory (or navigate there)
3. Exit any running autokey instance
3. run `python3 -m autokey.gtkui -l -c` from within the `lib` directory to start the version containing the potential fix from the source code.
4. Open the About dialogue in the help menu and verify you are on version `0.95.4+bugfix1_issue_225_and_226`
5. Try to reproduce the locking.

**Comment by herrdeh** [comment]
Wow - that's really instant delivery.
On the first glance, it seems to work nicely. I'll keep it under observation and report back in a few days!
Great, man!

**Comment by luziferius** [comment]
If you install this branch using pip3, remember to uninstall any packages installed using your distribution’s package manager. Otherwise the packages might mix. Then you end up with the launcher from this branch importing and running the other installation.

**Comment by luziferius** [comment]
The fix is in the master branch and will be part of the next release.

**Comment by lkorba** [comment]
I had the same issue when using paste (Shift+Ins) mode, it seems to be resolved now with this fix. Thanks! Looking forward for the next release.

**Comment by luziferius** [comment]
Released version [`0.95.5`](https://github.com/autokey/autokey/releases/tag/v0.95.5), containing the bug fix.
